lib/grape/dsl/settings.rb

```diff
@@ -171,7 +168,7 @@ def within_namespace(&_block)
       # the superclass's :inheritable_setting.
       def build_top_level_setting
         Grape::Util::InheritableSetting.new.tap do |setting|
-          if defined?(superclass) && superclass.respond_to?(:inheritable_setting) && superclass != Grape::API
+          if defined?(superclass) && superclass.respond_to?(:inheritable_setting) && superclass != Grape::API::Instance
```

Comment:
Is there a bug fix here?

Reply:
Oh, I forgot to mention this one. [Grape::API](https://github.com/ruby-grape/grape/blob/master/lib/grape/api.rb) doesn't make use of `Grape::Util::InheritableSetting`, but [Grape::API::Instance does](https://github.com/ruby-grape/grape/blob/master/lib/grape/api/instance.rb#L10).

```ruby
irb(main):006:0> Grape::API.respond_to?(:inheritable_setting)
=> false
irb(main):007:0> Grape::API::Instance.respond_to?(:inheritable_setting)
=> true
```

Thus, there is no reason to expect `Grape::API`. 

The condition was added when [Grape::API worked](https://github.com/ruby-grape/grape/blob/6244ff8b5cdac97bf949aa0f09d9d13967b1ac8e/lib/grape/api.rb#L7) with `Grape::Util::InheritableSetting`.

Reply:
I understand that part, but without digging into details it's a code change has a side effect. So either we're missing a spec or the entire `if` is unnecessary?

Reply:
It didn't cause big harm, however, a test for catching the regression makes sense :+1: Please, have a look! :slightly_smiling_face: